helix-term/src/ui/editor.rs

```diff
@@ -830,17 +830,23 @@ impl EditorView {
             last_mode = current_mode;
         };
 
-        match &key_result {
+        match key_result {
             KeymapResult::Matched(command) => {
-                execute_command(command);
+                execute_command(&command);
             }
             KeymapResult::Pending(node) => cxt.editor.autoinfo = Some(node.infobox()),
             KeymapResult::MatchedSequence(commands) => {
                 for command in commands {
-                    execute_command(command);
+                    execute_command(&command);
                 }
             }
-            KeymapResult::NotFound | KeymapResult::Cancelled(_) => return Some(key_result),
+            KeymapResult::NotFound => return Some(key_result),
+            KeymapResult::Cancelled(mut pending) => {
+                if !matches!(self.keymaps.get(mode, event), KeymapResult::NotFound) {
```

Comment:
I meant that you should recursively call `handle_keymap_event` here. Just calling `keymapsmget` directly misses a bunch of important logic